packages/typegpu/src/core/slot/accessor.ts

```diff
@@ -58,6 +59,34 @@ export class TgpuAccessorImpl<T extends AnyWgslData>
     this[$getNameForward] = this.slot;
   }
 
+  get [$gpuValueOf](): InferGPU<T> {
+    return new Proxy({
+      [$internal]: true,
+      [$ownSnippet]: this.#createSnippet(),
+      [$resolve]: (ctx) => ctx.resolve(this),
+      toString: () => `accessor:${getName(this) ?? '<unnamed>'}.$`,
+    }, valueProxyHandler) as InferGPU<T>;
+  }
+
+  /**
+   * @returns A snippet representing the accessor.
+   */
+  #createSnippet() {
+    // biome-ignore lint/style/noNonNullAssertion: it's there
+    const ctx = getResolutionCtx()!;
+    const value = getGpuValueRecursively(ctx.unwrap(this.slot));
+
+    if (isBufferUsage(value) || isBufferShorthand(value)) {
+      return snip(value, this.schema);
+    }
+
+    if (isTgpuFn(value)) {
+      return value[$internal].gpuImpl();
+    }
```

Comment:
Also could we not make the gpu value of tgpu fn its gpu impl? Then it would be unified
```suggestion
    const ctx = getResolutionCtx()!;
    const value = getGpuValueRecursively(ctx.unwrap(this.slot));

    if (isTgpuFn(value)) {
      return value[$internal].gpuImpl();
    }
```